Sources/Optimizely+Decide/OptimizelyUserContext.swift

```diff
@@ -61,6 +143,55 @@ public struct OptimizelyUserContext {
     }
 }
 
+extension OptimizelyUserContext {
+    
+    func getDecisionVariableMap(feature: FeatureFlag,
+                                variation: Variation?,
+                                enabled: Bool,
+                                reasons: DecisionReasons) -> [String: Any] {
+        var variableMap = [String: Any]()
+        
+        for (_, v) in feature.variablesMap {
+            var featureValue = v.value
+            if enabled, let variable = variation?.getVariable(id: v.id) {
+                featureValue = variable.value
+            }
+            
+            var valueParsed: Any? = featureValue
+            
+            if let valueType = Constants.VariableValueType(rawValue: v.type) {
+                switch valueType {
+                case .string:
+                    break
+                case .integer:
+                    valueParsed = Int(featureValue)
+                case .double:
+                    valueParsed = Double(featureValue)
+                case .boolean:
+                    valueParsed = Bool(featureValue)
+                case .json:
+                    valueParsed = OptimizelyJSON(payload: featureValue)?.toMap()
+                }
+            }
+            
+            if let value = valueParsed {
+                variableMap[v.key] = value
+            } else {
+                let info = OptimizelyError.variableValueInvalid(v.key)
+                logger.e(info)
+                reasons.addError(info)
+            }
+        }
+        
+        return variableMap
+    }
+    
+    func getAllOptions(with options: [OptimizelyDecideOption]?) -> [OptimizelyDecideOption] {
+        return (optimizely?.defaultDecideOptions ?? []) + (options ?? [])
```

Comment:
this could be confusing that default and options are added.  I see why, but, maybe a comment or something about it.